Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs

```diff
@@ -160,14 +248,34 @@ public static void GracefulProcessShutdown ([NotNull] Process process, TimeSpan
         // https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill#System_Diagnostics_Process_Kill
         // This exception can be swallowed safely, as we are sure that a process is associated and the process has
         // already exited, therefore the method can return at this point.
-        return;
       }
+    }
 
-      // No exception is thrown when calling .WaitForExit(..) on an already exited process.
-      if (process.WaitForExit (timeoutInMilliseconds))
-        return;
+    /// <summary>
+    /// Waits for the specified <paramref name="processes"/> to exit or until the <paramref name="sharedTimeout"/> is reached.
+    /// </summary>
+    /// <param name="processes">The processes that should be waited on to exit.</param>
+    /// <param name="sharedTimeout">The wait timeout after which waiting will be canceled.</param>
+    /// <returns>A subset of the specified <paramref name="processes"/> that did not exit withing the <paramref name="sharedTimeout"/>.</returns>
+    [MustUseReturnValue]
+    private static List<Process> WaitForProcessesExits (IEnumerable<Process> processes, TimeSpan sharedTimeout)
```

Comment:
Return type should be `IReadOnlyCollection<>` here imo. No use of returning a mutable collection, an immutable one makes more sense here.